patches/minecraft/net/minecraft/world/gen/layer/BiomeLayer.java.patch

```diff
@@ -0,0 +1,79 @@
+-   private static final int[] field_202746_t = new int[]{field_202730_d, field_202729_c, field_202741_o, field_202736_j};
+-   private static final int[] field_202747_u = new int[]{field_202731_e, field_202731_e, field_202731_e, field_202742_p};
+-   private int[] field_151623_c = field_202744_r;
++   private java.util.List<net.minecraftforge.common.BiomeManager.BiomeEntry>[] biomes = new java.util.ArrayList[net.minecraftforge.common.BiomeManager.BiomeType.values().length];
+-   public BiomeLayer(boolean p_i232147_1_) {
+-      if (p_i232147_1_) {
+-         this.field_151623_c = field_202743_q;
++   public BiomeLayer(boolean p_i232147_1_)
++   {
++      for (net.minecraftforge.common.BiomeManager.BiomeType type : net.minecraftforge.common.BiomeManager.BiomeType.values())
++      {
++         com.google.common.collect.ImmutableList<net.minecraftforge.common.BiomeManager.BiomeEntry> biomesToAdd = net.minecraftforge.common.BiomeManager.getBiomes(type);
++         int idx = type.ordinal();
++
++         if (biomes[idx] == null) biomes[idx] = new java.util.ArrayList<net.minecraftforge.common.BiomeManager.BiomeEntry>();
++         if (biomesToAdd != null) biomes[idx].addAll(biomesToAdd);
+       }
++      int desertIdx = net.minecraftforge.common.BiomeManager.BiomeType.DESERT.ordinal();
++
++      biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76769_d, 30));
++      biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_150588_X, 20));
++      biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76772_c, 10));
++
++      if (p_i232147_1_)
++      {
++         biomes[desertIdx].clear();
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76769_d, 10));
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76767_f, 10));
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76770_e, 10));
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76780_h, 10));
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76772_c, 10));
++         biomes[desertIdx].add(new net.minecraftforge.common.BiomeManager.BiomeEntry(Biomes.field_76768_g, 10));
++      }
+    }
+    public int func_202726_a(INoiseRandom p_202726_1_, int p_202726_2_) {
+@@ -46,21 +63,21 @@
+                return p_202726_1_.func_202696_a(3) == 0 ? field_202733_g : field_202734_h;
+             }
+-            return this.field_151623_c[p_202726_1_.func_202696_a(this.field_151623_c.length)];
++            return Registry.field_212624_m.func_148757_b(getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.BiomeType.DESERT, p_202726_1_).biome);
```

Comment:
I am not actually positive on the stance of Forge Registry vs Vanilla Registry use in patches, but is there a reason this is using the vanilla registry instead of the Forge Registry for biomes?

Reply:
There is not a reason, again it was an adaptation.

Reply:
Looking at it closer, it appears that the forge registries do not have an option to get the numerical id of a biome, correct me if I'm wrong.  This is why vanilla registries are required here.

Reply:
They... do, but that requires a cast to the actual underlying type (ForgeRegistry) rather than the field type of IForgeRegistry

Reply:
Ah, as such?:
`return ((ForgeRegistry) ForgeRegistries.BIOMES).getID(getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.BiomeType.ICY, context).biome);`
I suppose I should replace it with this?  Let me know what you think

Reply:
> and I do not prefer the ForgeRegistry cast hack
> as it ties things to one implementation
> - tterrag